### DISCUSSION TITLE
Memory management

### DISCUSSION THREAD

**Discussion by darrepac** [TomWright/dasel#358] [General]
Hello,

I am playing with big JSON files (about 1GB each) that I want to transform to CSV.
It works well under windows but when I move to Ubuntu, I have out of memory error. I have much less RAM in my Ubuntu device, but isn't an option in Dasel to not load all the file in memory to read it?

thanks

**Comment by darrepac** [comment]
I have the same issue with a virtual machine (Ubuntu) under windows

**Comment by TomWright** [comment]
I'm afraid dasel has to load the entire file into memory for it work. I'd be happy to address this in the future but I don't have the time right now as it would be a large amount of work.

**Reply by darrepac** [reply]
Yes not that easy to change memory management.
I end up using jq
